Tests/MetricsTests/TestMetrics.swift

```diff
@@ -135,17 +139,33 @@ internal class TestRecorder: RecorderHandler, Equatable {
 internal class TestTimer: TimerHandler, Equatable {
     let id: String
     let label: String
+    let storageUnit: TimeUnit
     let dimensions: [(String, String)]
 
     let lock = NSLock()
     var values = [(Date, Int64)]()
 
-    init(label: String, dimensions: [(String, String)]) {
+    init(label: String, storageUnit: TimeUnit, dimensions: [(String, String)]) {
         self.id = NSUUID().uuidString
         self.label = label
+        self.storageUnit = storageUnit
         self.dimensions = dimensions
     }
 
+    func retrieveValue(atIndex i: Int) -> Int64 {
```

Comment:
move to an extension defined on CoreMetricsTests.swift

Reply:
Would do, but the lock is inaccessible. I did move the TestCase to `CoreMetricsTests.swift`

Reply:
I see; placing is fine, though naming is a bit misleading; should this be `retriveInPreferredUnit`?

Reply:
I will update that!